jdaviz/configs/specviz2d/tests/test_parsers.py

```diff
@@ -134,6 +137,25 @@ def test_2d_1d_parser(specviz2d_helper, mos_spectrum2d, spectrum1d):
     specviz2d_helper.load_data(spectrum_2d=mos_spectrum2d, spectrum_1d=spectrum1d)
     assert specviz2d_helper.app.data_collection.labels == ['Spectrum 2D', 'Spectrum 1D']
 
+    spec2d_viewer = specviz2d_helper.app.get_viewer("spectrum-2d-viewer")
+    assert spec2d_viewer.figure.axes[0].label == "x: pixels"  # -0.5, 14.5
+    spec2d_viewer.apply_roi(XRangeROI(10, 12))
+
+    spec2d_viewer.session.edit_subset_mode._mode = NewMode
+
+    spec1d_viewer = specviz2d_helper.app.get_viewer("spectrum-viewer")
+    assert spec1d_viewer.figure.axes[0].label == "Wavelength [Angstrom]"  # 6000, 8000
+    spec1d_viewer.apply_roi(XRangeROI(7000, 7100))
+
+    # Subset 1 should follow Spectrum 2D viewer unit.
+    # Subset 2 should follow Spectrum 1D viewer unit.
+    subsets = specviz2d_helper.app.get_subsets()
+    assert len(subsets) == 2
+    s1 = subsets["Subset 1"]
+    s2 = subsets["Subset 2"]
+    assert s1.lower.unit == s1.upper.unit == u.pix

```

Comment:
I confirmed that this line throws exception on `main` without this patch, but passes with this patch. Exception on `main`:

```
>       assert s1.lower.unit == s1.upper.unit == u.pix
E       assert Unit("Angstrom") == Unit("pix")
E        +  where Unit("Angstrom") = <Quantity 12. Angstrom>.unit
E        +    where <Quantity 12. Angstrom> = Spectral Region, 1 sub-regions:\n  (10.0 Angstrom, 12.0 Angstrom) .upper
E        +  and   Unit("pix") = u.pix
```